blobfuse2-perf.yaml

```diff
@@ -0,0 +1,140 @@
+schedules:
+  # Cron string < minute hour day-of-month month day-of-week>
+  #             * means all like '*' in day of month means everyday
+  # Run only on main branch
+  # 'always' controls whether to run only if there is a change or not
+  # Run this pipeline every 15:00 time
+  - cron: '0 15 * * 6'
+    displayName: 'Weekly Perf Blobfuse2 Perf Run'
+    branches:
+      include:
+        - main
+
+jobs:
+  - job: Blobfuse2_Perf_Test
+    timeoutInMinutes: 2800 # two day timeout
+    strategy:
+      matrix:
+        Ubuntu-20:
+          DistroVer: "Ubn20_PERF"
+          AgentName: "UBN20-PERF"
+          Description: "Blobfuse2 Perf Test"
+
+    pool:
+      name: "Blobfuse Pool"
+      demands:
+        - Agent.Name -equals $(AgentName)
+
+    variables:
+      - group: NightlyBlobFuse
+      - name: MOUNT_DIR
+        value: "/home/vsts/workv2/blob_mnt"
+      - name: TEMP_DIR
+        value: "/home/vsts/workv2/blobfuse2tmp"
+      - name: BLOBFUSE2_CFG
+        value: "/home/tamer/blobfuse2.yaml"
+      - name: GOPATH
+        value: "/home/vsts/workv2/go"
+      - name: ROOT_DIR
+        value: "/home/vsts/workv2/"
+      - name: WORK_DIR
+        value: "/home/vsts/workv2/go/src/azure-storage-fuse"
+
+    steps:
+      - checkout: none
+
+        # Prestart cleanup
+      - script: |
+          sudo fusermount -u $(MOUNT_DIR)
+          sudo kill -9 `pidof blobfuse2`
+          sudo rm -rf $(ROOT_DIR)
+        displayName: 'PreBuild Cleanup'
+
+      # Create directory structure
+      - script: |
+          sudo rm -rf $(ROOT_DIR)
+          sudo mkdir -p $(ROOT_DIR)
+          sudo chown -R `whoami` $(ROOT_DIR)
+          chmod 777 $(ROOT_DIR)
+          mkdir -p $(ROOT_DIR)/go/src
+        displayName: 'Create Directory Structure'
+  
+      # Checkout the code  
+      - script: |
+          git clone https://github.com/Azure/azure-storage-fuse
+        displayName: 'Checkout Code'
+        workingDirectory: $(ROOT_DIR)/go/src
+
+      # Pull the branch
+      - script: |
+          git checkout `echo $(Build.SourceBranch) | cut -d "/" -f 1,2 --complement`
+        displayName: 'Checkout Branch'
+        workingDirectory: $(WORK_DIR)
+
+      # -------------------------------------------------------
+      # Pull and build the code
+      - template: 'azure-pipeline-templates/build.yml'
+        parameters:
+          working_directory: $(WORK_DIR)
+          root_dir: $(ROOT_DIR)
+          mount_dir: $(MOUNT_DIR)
+          temp_dir: $(TEMP_DIR)
+          gopath: $(GOPATH)
+          container: cont1
+          skip_ut: true
+
+      - script: |
+          cd $(WORK_DIR)
+          $(WORK_DIR)/blobfuse2 gen-test-config --config-file=azure_key.yaml --container-name=cont1 --temp-path=$(TEMP_DIR) --output-file=$(BLOBFUSE2_CFG)
```

Comment:
Here we are comparing v2 preview build vs v2 main branch. Shall we run the same test with v1 binary also. Just to compare how much deviation we are having from there as well. Relying on old numbers of v1 may not be a good idea because based on network/storage/azure and all factors these numbers may get skewed some times. At then end we can generate a report showing v1 numbers, preview binary numbers and main branch numbers.

Reply:
We can add v1 in a separate pr - for now let's try to get this in